birdyboard.py

```diff
@@ -0,0 +1,421 @@
+
+    """
+
+    def __init__(self):
+        # construction of three subclasses.
+        self.chirper = Chirper()
+        self.threader = Threader()
+        self.usurper = Usurper()
+
+        self.user_name = ""
+        self.user_id = ""
+
+        self.thread_id = ""
+
+        self.public_or_private = "public"
+
+# ###############################
+# ######## MENU PRINTERS ########
+# ###############################
+
+        self.unlogged_in_options = "Welcome. You are not logged in.\n1. new user\n2. log in\n3. view public chirps\n'x' to exit.\n"
+        self.logged_in_options = "1. log out\n2. view public chirps\n3. view private chirps\n4. New public chirp thread\n5. new private chirp thread\n'x' to exit.\n"
+
+# ############################################
+# ######## UNLOGGED IN TOP LEVEL MENU ########
+# ############################################
+
+    def unlogged_in_menu(self):
+        """
+        requests input for top level menu functionality while the user is not logged in. Handles whether the user would like to log in, create a user, exit, or view public chirps.
+        Arguments: none
+        """
+        next_step = input(">> ")
+        if next_step == "1":  # create a user.
+            print("creating a new user.")
+            self.create_a_user_menu()
+        elif next_step == "2":  # log in to a current user.
+            self.usurper.generate_users_list()
+            self.users_menu()
+        elif next_step == "3":  # view public chirps. Does not have the option of commenting.
+            self.public_or_private = "public"
+            self.threader.generate_public_threads_list()
+            self.view_threads_menu()
+        elif next_step == "x":  # exit.
+            print("goodbye.")
+            exit()
+        else:
+            print("command not found.")
+            self.unlogged_in_menu()
+
+# ##########################################
+# ######## LOGGED IN TOP LEVEL MENU ########
+# ##########################################
+
+    def logged_in_menu(self):
+        """
+        Top level menu to print when user logs in or when logged in user traverses back to top-level. requests a next-step input and handles whether the user would like to log out, exit, view a public chirp, view a private chirp, or create a new public or private thread.
+        Arguments: None
+        """
+        next_step = input(">> ")
+        if next_step == "1":  # log out
+            print('logging out.')
+            self.user_name = ""
+            print(self.unlogged_in_options)
+            self.unlogged_in_menu()
+        elif next_step == "2":  # view public threads.
+            self.public_or_private = "public"
+            self.threader.generate_public_threads_list()
+            self.view_threads_menu()
+        elif next_step == "3":  # view private threads.
+            self.public_or_private = "private"
+            self.threader.generate_private_threads_list(self.user_id)
```

Comment:
Yeah, never mind.   Looked at the actual method and its only output is printing.  The `generate_list` part threw me at first glance, expecting it to set `self.private_threads_list` or something.